### DISCUSSION TITLE
PE pattern seems to be broken

### DISCUSSION THREAD

**Discussion by r3verseer** [WerWolv/ImHex#470] [Pattern Language Q&A]
PE pattern seems to be broken... When I am trying to open DOS header, ImHex is crashing - probably problem with pointer.  Patterns shouldnt crash app in any circumstances - extensive error handling should be performed to avoid such situation

**Comment by WerWolv** [comment]
Writing a language is hard and comments like that don't really help. Even compilers like gcc or clang occasionally crash.
Best you can do is open an issue on the repo stating what happened (and in the best case submit a stack trace of the crash).

That said, the issue has been fixed a few commits back so you can use the latest nightly build.

**Reply by r3verseer** [reply]
Sorry - i have read my post and it seems quite rude, especially as you created my favourite hex editor - i did not mean anything wrong, What i have noticed is: when you  open PE pattern and try to select DOS header section app is crashing. I just wanted to suggest that ImHex should check pattern code before applyting to avoid crashes and potential security reason. I know error handling is really hard (as in this case we have PL writen inside other one so dont expect this to be fixed soon.

**Comment by birorobi** [comment]
Sorry.... yes is broken..  Lp.

pon, 4. tra 2022. u 23:08 ReverSeer ***@***.***> napisao je:
